### DISCUSSION TITLE
Use of bindings in a MySQL query

### DISCUSSION THREAD

**Discussion by diogenesbrussels** [Budibase/budibase#9198] [Help]
Hi!

I am confused about the use of bindings in a MySQL query.
How are they handled?
If I use the following query:
`SELECT my_column AS answer, COUNT(*) AS c FROM `test_situations` GROUP BY my_column LIMIT 30;`
It returns the expected result (list of different values with their count) but if I use a binding ({{current_col}} = my_column) in the query :
`SELECT {{current_col}} AS answer, COUNT(*) AS c FROM `test_situations` GROUP BY {{current_col}} LIMIT 30;`
I get only one result with the total count
`{
  "answer": "current_col",
  "c": 7898
}`

Why? How to solve this problem?
Thank you!

**Comment by melohagan** [comment]
Hey @diogenesbrussels 

Could you share some screenshots of your query config please?

**Comment by diogenesbrussels** [comment]
Hey @melohagan ,

Here it is:
![image](https://user-images.githubusercontent.com/91942877/210517027-cbdf7272-f6cc-4a2f-8bca-7a73bd7cf394.png)

Thank you!

**Reply by diogenesbrussels** [reply]
![image](https://user-images.githubusercontent.com/91942877/210529216-208d6da1-9a60-48ce-9745-f99990434dab.png)

**Reply by melohagan** [reply]
I think what's happening here is that the value `my_column` is being wrapped in quotes which is interfering with the query. I tested this on SQL Server and it throws an error. 

I don't think anything can be done about this because it was added for security reasons. This is also an issue that users have come across when trying to use a dynamic table name.

**Reply by diogenesbrussels** [reply]
Hey @melohagan ,

Thanks for your feedback, it's really a bit sad as I wanted to integrate table management tools within budibase for some of my users. I would have liked to find a solution with budibase without having to use external tools.

Would it be possible, to make this clearer, for a preview of the generated query to be displayed below the edit field? This would help to debug similar situations, I have also had this happen with dynamic table names.

Thank you!

**Reply by diogenesbrussels** [reply]
Or something like this :
![image](https://user-images.githubusercontent.com/91942877/211528927-5b62efc5-c608-48b5-93ef-74fc4e3977b1.png)
